client/src/pages/Project.tsx

```diff
@@ -53,16 +56,57 @@ const Button = styled(BaseButton)`
     background-color: rgb(59, 130, 246);
 `;
 
+const Button2 = styled(Button)`
+    width: 45%;
+`;
+
+const ButtonDeleteProject = styled(BaseButton)`
+    background-color: rgb(255, 0, 90);
+    width: 45%;
+`;
+
+const ButtonLeaveProject = styled(BaseButton)`
+    background-color: rgb(255, 0, 90);
+`;
+
+const ButtonConfirmDelete = styled(BaseButton)`
+    background-color: rgb(255, 0, 90);
+    border: 1px solid rgb(255, 0, 90);
+    float: centre;
+    margin-left: 0.75rem;
+    margin-top: 1rem;
+`;
+
+const ButtonConfirmCancel = styled(BaseButton)`
+    background-color: white;
+    border: 1px solid black;
+    color: black;
+    float: centre;
+    margin-top: 1rem;
+    margin-left: 0.75rem;
+`;
+
 const Project = (): JSX.Element => {
     const auth = useContext(AuthContext);
     const data = useContext(DataContext);
+    const navigate = useNavigate();
 
     const { id: projectid } = useParams();
     const [loading, setLoading] = useState<boolean>(true);
     const [project, setProject] = useState<IProject | undefined>(undefined);
     const [tasks, setTasks] = useState<ITask[]>([]);
     const [showInviteWindow, setShowInviteWindow] = useState<boolean>(false);
     const isAdmin: boolean = project?.members.find((u) => u.name === auth.auth.user)?.role === "admin";
+    const [newName, setNewName] = useState<string>();
+    const [newDesc, setNewDesc] = useState<string>();
+    const [isPublic, setPublic] = useState<boolean>();
+    const [currUsers, setCurrUsers] = useState<Array<IUser>>([]);
+    const [removedUsers, setRemovedUsers] = useState<Array<IUser>>([]);
+
+    const [showSettingsModal, setShowSettingsModal] = useState<boolean>(false);
+    const [showDeleteModal, setShowDeleteModal] = useState<boolean>(false);
```

Comment:
feels like too many modals?

might be better to make project_settings its own page if you want it to summon modals, cos Modal.tsx doesn't really handle multiple modals well -> unless want to change Modal.tsx to take in a stack (or some other data structure) of jsx elements and push/pop them so can click outside/press esc properly (which sounds like it can fix current problem)

also, maybe a cancel/close button for the main modal too